service/telemetry.go

```diff
@@ -179,6 +165,37 @@ func (tel *telemetryInitializer) initOnce(buildInfo component.BuildInfo, logger
 	return nil
 }
 
+func buildTelAttrs(buildInfo component.BuildInfo, cfg telemetry.Config) map[string]string {
```

Comment:
Separate concern, but we should look into that. When we "restart" the telemetry due to for example a config update, we generate a new instance ID which I think we should not? Let's open an issue to discuss that.

Reply:
Ideally, yes, the instance ID should persist. Issue created https://github.com/open-telemetry/opentelemetry-collector/issues/6158

Reply:
Thanks